### PR TITLE
[Windows] Update CollectionView changing ItemsLayout 

### LINKED ISSUE #7747: CollectionView does not update when changing ItemsLayout
### Description

I have a CollectionView with several items and I want to make a 2 column layout on wide screens and a 1 column layout on tall screens.

I tried setting a GridItemsLayout and changing the Span, but that did not work and then I tried changing the entire ItemsLayout, but still no updates in the app.

Restarting the app works, so this is just the result of changing at runtime.

### Steps to Reproduce

Add a CollectionView with items, and then say write this code to toggle the layout on page size changes:

```cs
private void OnSizeChanged(object? sender, EventArgs e) =>
    collectionView.ItemsLayout = Width > 600
        ? new GridItemsLayout(2, ItemsLayoutOrientation.Vertical)
        : new LinearItemsLayout(ItemsLayoutOrientation.Vertical);
```

> NOTE: this should all be doable with AdaptiveTriggers, but that is not working currently #7242

### Version with bug

6.0 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

Windows

### Affected platform versions

All

### Did you find any workaround?

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by jsuarezruiz**
### Description of Change

Update CollectionView changing ItemsLayout on Windows.

fix-7747

To test the changes, launch the .NET MAUI Gallery on Windows and navigate to the CollectionView samples. Select the added "Adaptive CollectionView" sample. This sample change the CollectionView  ItemsLayout property based on the Page Width. Resize the Window to validate that the CollectionView updates the layout.

### Issues Fixed

Fixes (but reverted) #7747

### CODE REVIEW

File: src/Controls/src/Core/Handlers/Items/ItemsViewHandler.Windows.cs [hunk 1]
```diff
@@ -279,18 +283,40 @@ protected virtual void UpdateEmptyView()
...
-			_defaultHorizontalScrollVisibility = null;
-			_defaultVerticalScrollVisibility = null;
+					if (linearItemsLayout.ItemSpacing != _previousItemSpacing)
+					{
+						_previousItemSpacing = linearItemsLayout.ItemSpacing;
+						gridView.ItemContainerStyle = linearItemsLayout.GetItemContainerStyle();
+					}
+				}
 
-			UpdateItemTemplate();
```
Comment (rmarinho) [inline src/Controls/src/Core/Handlers/Items/ItemsViewHandler.Windows.cs:289]: why is this removed? were is It being called now?

File: src/Controls/src/Core/Handlers/Items/ItemsViewHandler.Windows.cs [hunk 2]
```diff
@@ -32,6 +32,10 @@ public abstract partial class ItemsViewHandler<TItemsView> : ViewHandler<TItemsV
 		bool _emptyViewDisplayed;
 		double _previousHorizontalOffset;
 		double _previousVerticalOffset;
+		double _previousItemSpacing;
```
[the remark this answers is not in the document]
Comment (jsuarezruiz) [reply src/Controls/src/Core/Handlers/Items/ItemsViewHandler.Windows.cs:35]: To avoid to recreate and inject the ItemContainerStyle if the spacing not changed. For example, resizing the Window. In this case, without checking the changes will change the ItemContainerStyle once every time the size changes.

File: src/Controls/src/Core/Handlers/Items/ItemsViewHandler.Windows.cs (hunk 1 above)
Comment (jsuarezruiz) [reply src/Controls/src/Core/Handlers/Items/ItemsViewHandler.Windows.cs:289]: With the PR changes nowhere, check [this comment](https://github.com/dotnet/maui/pull/14532/files#r1164070864) and let me know if you have any questions.

File: src/Controls/src/Core/Handlers/Items/ItemsViewHandler.Windows.cs [hunk 3]
```diff
@@ -279,18 +283,40 @@ protected virtual void UpdateEmptyView()
...
-			UpdateItemTemplate();
-			UpdateItemsSource();
-			UpdateVerticalScrollBarVisibility();
-			UpdateHorizontalScrollBarVisibility();
-			UpdateEmptyView();
+				if (Layout is GridItemsLayout gridItemsLayout)
+				{
+					gridView.Orientation = gridItemsLayout.Orientation == ItemsLayoutOrientation.Horizontal
+						? Orientation.Horizontal
+						: Orientation.Vertical;
```
Comment (mandel-macaque) [inline src/Controls/src/Core/Handlers/Items/ItemsViewHandler.Windows.cs:307]: WE have code duplication here, add an extension method that does the ItemsLayoutOrientation to Orientation conversion, probably a good API to have to be used internally.

File: src/Controls/src/Core/Handlers/Items/ItemsViewHandler.Windows.cs [hunk 4]
```diff
@@ -32,6 +32,10 @@ public abstract partial class ItemsViewHandler<TItemsView> : ViewHandler<TItemsV
 		bool _emptyViewDisplayed;
 		double _previousHorizontalOffset;
 		double _previousVerticalOffset;
+		double _previousItemSpacing;
+		double _previousHorizontalItemSpacing;
+		double _previousVerticalItemSpacing;
```
Comment (mandel-macaque) [inline src/Controls/src/Core/Handlers/Items/ItemsViewHandler.Windows.cs:37]: Are all those set at the same time, at some point it might be a good idea to create a private structure to store this and make the code cleaner.

File: src/Controls/src/Core/Handlers/Items/ItemsViewHandler.Windows.cs [hunk 5]
```diff
@@ -279,18 +283,40 @@ protected virtual void UpdateEmptyView()
...
 		protected virtual void UpdateItemsLayout()
 		{
-			ListViewBase.IsSynchronizedWithCurrentItem = false;
+			if (ListViewBase is FormsGridView gridView)
+			{
+				if (Layout is LinearItemsLayout linearItemsLayout)
+				{
+					gridView.Orientation = linearItemsLayout.Orientation == ItemsLayoutOrientation.Horizontal
+						? Orientation.Horizontal
+						: Orientation.Vertical;
```
Comment (mandel-macaque) [inline src/Controls/src/Core/Handlers/Items/ItemsViewHandler.Windows.cs:292]: Duplicated code.

File: src/Controls/src/Core/Platform/Windows/Extensions/CollectionViewExtensions.cs [hunk 6]
```diff
@@ -0,0 +1,40 @@
+﻿using Microsoft.UI.Xaml;
+using Microsoft.UI.Xaml.Controls;
+using WSetter = Microsoft.UI.Xaml.Setter;
+using WStyle = Microsoft.UI.Xaml.Style;
+
+namespace Microsoft.Maui.Controls.Platform
+{
+	public static class CollectionViewExtensions
+	{
+		public static WStyle GetItemContainerStyle(this LinearItemsLayout layout)
```
Comment (mandel-macaque) [inline src/Controls/src/Core/Platform/Windows/Extensions/CollectionViewExtensions.cs:10]: Do we want to be able to return something that is not null for a null layout? Does the following code make sense:
```csharp
var test = (LinearItemsLayout) null;
var style = test.GetItemContainerStyle();
```

Style won't be null yet we have a null object, that is confusing.

File: src/Controls/src/Core/Platform/Windows/Extensions/CollectionViewExtensions.cs [hunk 7]
```diff
@@ -0,0 +1,40 @@
+﻿using Microsoft.UI.Xaml;
+using Microsoft.UI.Xaml.Controls;
+using WSetter = Microsoft.UI.Xaml.Setter;
+using WStyle = Microsoft.UI.Xaml.Style;
```
Comment (mandel-macaque) [inline src/Controls/src/Core/Platform/Windows/Extensions/CollectionViewExtensions.cs:4]: '#nullable enabled' ? If it is the default, I am surprise we are getting no warnings.

**Review by mandel-macaque — CHANGES_REQUESTED** [review]

File: src/Controls/src/Core/Handlers/Items/ItemsViewHandler.Windows.cs (hunk 5 above)
Comment (jsuarezruiz) [reply src/Controls/src/Core/Handlers/Items/ItemsViewHandler.Windows.cs:292]: Updated to use a created `ToPlatform` extension method.

**Review by rmarinho — APPROVED** [review]

**Review by mandel-macaque — APPROVED** [review]

**Comment by Foda** [comment]
FYI: this PR broke https://github.com/dotnet/maui/pull/14391

**Comment by sjorsmiltenburg** [comment]
Can this fix be merged into an update for 7.x? It's a blocking issue for my app.

**Comment by sjorsmiltenburg** [comment]
@samhouts 
I read above that this merge is partially reverted, but the status of the issue is still 'merged'.
One might conclude that only some changes are reverted and the issue is still resolved.
I want to make clear that this is not the case and the bug of not listening to ItemsLayout changes is now back when moving from 8.0.0-rc.2.9530 -> 8.0.0-rc.29530
Should this issue be re-set to 'open'?
